src/views/viewer/ChannelView/ChannelView.tsx

```diff
@@ -83,15 +141,61 @@ export const ChannelView: React.FC = () => {
             )}
           </TitleContainer>
           <StyledButtonContainer>
-            <Button variant={isFollowing ? 'secondary' : 'primary'} onClick={handleFollow} size="large">
+            <StyledButton
+              icon={isFollowing ? <SvgGlyphCheck /> : <SvgGlyphPlus />}
+              variant={isFollowing ? 'secondary' : 'primary'}
+              onClick={handleFollow}
+              size="large"
+            >
               {isFollowing ? 'Unfollow' : 'Follow'}
-            </Button>
+            </StyledButton>
           </StyledButtonContainer>
         </TitleSection>
-      </Header>
-      <VideoSection className={transitions.names.slide}>
-        <InfiniteVideoGrid channelId={id} showChannel={false} />
-      </VideoSection>
+        <TabsContainer>
+          <Tabs initialIndex={0} tabs={mappedTabs} onSelectTab={handleSetCurrentTab} />
+          {currentTabName === 'Videos' && (
+            <SortContainer>
+              <Text variant="body2">Sort by</Text>
+              <Select helperText={null} value={sortVideosBy} items={SORT_OPTIONS} onChange={handleSorting} />
+            </SortContainer>
+          )}
+        </TabsContainer>
+        {(() => {
+          switch (currentTabName) {
+            case 'Videos':
+              return (
+                <>
+                  <VideoSection className={transitions.names.slide}>
+                    <Grid maxColumns={null} onResize={handleOnResizeGrid}>
+                      {videosWithPlaceholders.map((video, idx) => (
+                        <VideoPreview key={idx} id={video.id} showChannel={false} />
+                      ))}
+                    </Grid>
+                  </VideoSection>
+                  <PaginationContainer>
+                    <Pagination
+                      onChangePage={handleChangePage}
+                      page={currentPage}
+                      itemsPerPage={videosPerPage}
+                      totalCount={totalCount}
+                    />
+                  </PaginationContainer>
+                </>
+              )
+            case 'About':
+              return <ChannelAbout />
+          }
+        })()}
```

Comment:
We shouldn't use a pattern like that, with invoking an anonymous function right away. Let's move that condition out of `return` entirely

Reply:
I think is a clean way to not have too many conditional operators, moving it out is also fine